tests/test_core.py

```diff
@@ -53,7 +53,10 @@ def test_device_failed_properties(self):
         store.GetValue = mock.Mock(side_effect=_ctypes.COMError(None, None, None))
         dev.OpenPropertyStore = mock.Mock(return_value=store)
         with warnings.catch_warnings(record=True) as w:
-            AudioUtilities.CreateDevice(dev)
+            device = AudioUtilities.CreateDevice(dev)
+            # Properties are lazy-loaded and won't throw exceptions
+            # until accessed
+            dev = device.properties
         assert len(w) == 1
```

Comment:
Thanks for updating the tests too.
Maybe we should take the CreateDevice() call out of the context manager and only put the property access into it to prove that it's indeed raising only with this specific line.
So test would be:
```python
        AudioUtilities.CreateDevice(dev)
        device = AudioUtilities.CreateDevice(dev)
        with warnings.catch_warnings(record=True) as w:
            # Properties are lazy-loaded and won't throw exceptions
            # until accessed
            device.properties
        assert len(w) == 1
```